### ISSUE TITLE
Reproducing MoCo on ImageNet-1k.

### ISSUE THREAD

**Issue by yzou2** [HobbitLong/CMC#41]
Hi Yonglong,

Thanks a lot for the great work and sharing the code. I am trying to reproduce the results of MoCo on ImageNet-1k, with ResNet 50. Did you reproduce the results on Kaiming's paper on the full ImageNet? Would you kindly share me the specific configurations for reproducing MoCo-ResNet-50?

Thanks a lot!

**Comment by HobbitLong** [comment]
Hi, @yzou2,

Thanks for your kind words! I have reproduced his results  (60.6 with 64 negatives) on ImageNet. I strictly followed his pre-training recipe (8GPUs, 256 batch size, 200 epochs, blabla), and used a Distributed Dataparallel training code (I will release this once I got some time).

**Comment by yzou2** [comment]
Thanks a lot, Yonglong!

**Comment by happywu** [comment]
Hi @HobbitLong ,

Did you modify other parts than this distributed data parallel training (will this distributed training influence accuracy)? Could you please tell that If I directly change the configuration to (8Gpus, 256 batch size, 200 epochs) without changing current code, would this achieve comparable performance?

Thanks!

**Comment by HobbitLong** [comment]
Hi, @yzou2, @happywu,

Maybe this is a bit late, but just let you know that the DDP implementation for MoCo (as well as InstDis, MoCo v2, PIRL, ...) have been released [here](https://github.com/HobbitLong/PyContrast).

**Comment by yzou2** [comment]
> Hi, @yzou2, @happywu,
> 
> Maybe this is a bit late, but just let you know that the DDP implementation for MoCo (as well as InstDis, MoCo v2, PIRL, ...) have been released [here](https://github.com/HobbitLong/PyContrast).

It's really great! Thanks!
